### PR TITLE
 [Improve](query) improve column match performance by introducing a column name map in `MaterializedIndexMeta`

### PR DESCRIPTION
**Pull request by eldenmoon**
# Proposed changes

`getVisbleColumn` is slow due to the linear search process, using a map to speed up search.

image

270列select * limit 1
200qps -> 1000qps

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.05 seconds
 stream load tsv:          448 seconds loaded 74807831229 Bytes, about 159 MB/s
 stream load json:         25 seconds loaded 2358488459 Bytes, about 89 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230329065044_clickbench_pr_122135.html

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/MaterializedIndexMeta.java [hunk 1]
```diff
@@ -215,13 +220,14 @@ public static boolean matchColumnName(String lhs, String rhs) {
         return normalizeName(lhs).equalsIgnoreCase(normalizeName(rhs));
     }
 
+    public Column getColumnByDefineName(String colDefineName) {
+        String normalizedName = normalizeName(colDefineName);
+        return definedNameToColumn.getOrDefault(normalizedName, null);
+    }
+
     public Column getColumnByName(String columnName) {
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/MaterializedIndexMeta.java:228]: yes! I shohuld add String.CASE_INSENSITIVE_ORDER to map

[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/MaterializedIndexMeta.java:228]: done

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**